coalib/settings/Section.py

```diff
@@ -159,8 +202,13 @@ def __getitem__(self, item, ignore_defaults=False):
         if key == '':
             raise IndexError('Empty keys are invalid.')
 
-        res = self.contents.get(key, None)
+        res = copy.deepcopy(self.contents.get(key, None))
```

Comment:
oh you *always* want to return the copy making them immutable? Then why not remove the setter for the setting